src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -509,16 +509,18 @@ private function getShell()
 
     private function isTty(): bool
     {
-        if (!\defined('STDIN')) {
+        $inputStream = $this->inputStream ?: fopen('php://stdin', 'r');
+
+        if ('php://stdin' !== (stream_get_meta_data($inputStream)['url'] ?? null)) {
```

Comment:
actually this is not needed to fix the failure, but making the behavior of this method global looks suspicious to me so here a local check

Reply:
note that this check works even if `$inputStream === STDIN`

Reply:
Are you sure with `url` here?

```
>>> stream_get_meta_data(STDIN)['url']
PHP Notice:  Undefined index: url in phar://eval()'d code on line 1
=> null
>>> stream_get_meta_data(STDIN)
=> [
     "timed_out" => false,
     "blocked" => true,
     "eof" => false,
     "wrapper_type" => "PHP",
     "stream_type" => "STDIO",
     "mode" => "rb",
     "unread_bytes" => 0,
     "seekable" => false,
     "uri" => "php://stdin",
   ]
```

Reply:
Pretty much yes: https://3v4l.org/njZZV
What's your runtime?

Reply:
That 3v4l snippet follows behaviour I observe on my PC, it looks to be `uri`, not `url`

Reply:
:facepalm: